### DISCUSSION TITLE
Snackbar/Toast stuck inside the story container

### DISCUSSION THREAD

**Discussion by RemyMachado** [storybookjs/storybook#28922] [Help]
### Summary

Hey,

When I render my snackbar component it gets stuck inside the story container instead of being rendered on the bottom right of the page. Is there an approach to this issue?  

![stuck-snackbar](https://github.com/user-attachments/assets/7ebf231b-974f-480a-9bed-45c4ee13b9e0)  

I tried moving the SnackbarViewport (renderer) in the DOM directly under the `body`, but it's not working. The `viewport` element is correctly found, but moving it causes this error:  
`Uncaught DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.`

```typescript
  decorators: [
    Story => {
      useEffect(() => {
        const viewport = document.querySelector('[data-storybook-id="snackbar-viewport"]');
        
        document.body.appendChild(viewport);
      }, []);

      return (
        <>
          <Story />
          <SnackbarFactory />
        </>
      );
    },
  ],
  ```

Any help would be greatly appreciated,

**Comment by dschulten** [comment]
In addition, the decorator on the stories level decorates every component, so one gets an additional snackbar/toast for every story. If I have seven toast stories with toast variants, I get hammered with seven toasts all doing the same thing, one in each story ;-)
Hopefully there is a way to use autodoc and still achieve this.
